### ISSUE TITLE
Loss balancer

### ISSUE THREAD

**Issue by turian** [Lightning-AI/lightning#16466]
### Description & Motivation

It's commonplace to combine many losses (l1 loss, feature matching loss, multiscale losses, etc.). However, when they are on different scales, it is fiddly to choose the weighting. And, the weighting might change over time.

A loss balancer would be a useful addition, using EMA.

### Pitch

Meta's encodec has a loss balancer and it makes it very easy for them to mix many different scaled losses.

This would probably be an easy win for many lightning users that mix losses.

### Alternatives

There is available code for the balancer: https://github.com/facebookresearch/encodec/blob/main/encodec/balancer.py

However: a) the licensing is not permissive and b) it doesn't work in DDP.

### Additional context

https://arxiv.org/pdf/2210.13438.pdf

**Comment by carmocca** [comment]
Have you tried using their balancer implementation with manual optimization?

**Comment by turian** [comment]
@carmocca No, I think that's a decent workaround for now. Actually one can just compute the loss of each component over validation and use a fixed balancing scheme too.

**Comment by henryruhs** [comment]
any updates on this?
